packages/cli/src/daemon-config.ts

```diff
@@ -0,0 +1,285 @@
+import 'reflect-metadata'
+import { jsonObject, jsonMember, jsonArrayMember, TypedJSON, toJson } from 'typedjson'
+
+/**
+ * Whether the daemon should start its own bundled in-process ipfs node, or if it should connect
+ * over HTTP to an existing remote IPFS node.
+ */
+export enum IpfsMode {
+  BUNDLED = 'bundled',
+  REMOTE = 'remote',
+}
+
+/**
+ * Ceramic Daemon options for configuring behavior related to IPFS.
+ */
+@jsonObject
+@toJson
+export class DaemonIpfsConfig {
+  /**
+   * Whether the daemon should start its own bundled in-process ipfs node, or if it should connect
+   * over HTTP to an existing remote IPFS node.
+   */
+  @jsonMember(String)
```

Comment:
I'm disappointed at the need of these type specifiers in the decorators.  They're only there for tests.  The regular ceramic daemon compiles and runs just fine without them - the `typedjson` library uses `reflect-metadata` to infer the types on its own just fine.  But when running in jest with babel it couldn't do the type inference unless I put these here explicitly.